src/context/RecipeContext.jsx

```diff
@@ -47,102 +44,73 @@ export const RecipeProvider = ({children}) =>{
-    // URL/RECIPE STATES
-
-    // RANDOM NUM
+    // RANDOM NUM FOR INIT OFFSET
     const randomNum = ()=>{return Math.floor((Math.random() * 900))}
-    const [offset,setOffset] = useState(randomNum)
-    const [searchTerm,setSearchTerm] = useState('')
-    const [recipeType,setRecipeType] = useState('')
-    const [diet,setDiet] = useState('')
-    const [cuisine,setCuisine] = useState('')
-    const [intolorances,setIntolorances] = useState('')
-    const [noOfResults,setNoOfResults] = useState(8)
-    const [currentRecipe,setCurrentRecipe] = useState('')
-    const [url,setUrl] = useState(`https://api.spoonacular.com/recipes/complexSearch?query=${searchTerm}&number=${noOfResults}&offset=${offset}&cuisine=${cuisine}&diet=${diet}&intolorances=${intolorances}&type=${recipeType}&apiKey=033797df84694890b040b816a119b147`)
 
-    const writeUserData = (userId,favorites)=>{
-        const db = getDatabase()
-        set(ref(db, 'users/' + userId), {
-            favorites: favorites
-        });
-    }
-
-    const getUserData = (user)=>{
-            const db = getDatabase();
-            const favoritesRef = ref(db, 'users/' + user.uid);
-            onValue(favoritesRef, (snapshot) => {
-            const data = snapshot.val();
-            //snapshot.val returns data from database attatched to user.
-            //data.favorites = favorites array that was set from writeUserData()
-            data ? setUserFavorites(data.favorites) : setUserFavorites(null)
-        });
-    }
-
-    const handleWriteUserData = ()=>{
-            writeUserData(currentUser.uid,userFavorites)
-    }
+    // URL STATES
+    const [urlEndpoints,setUrlEndpoints] = useState({
+        searchTerm: '',
+        noOfResults: 8,
+        offset: randomNum(),
+        cuisine: '',
+        diet: '',
+        intolorances: '',
+        recipeType: '',
+    })
+    const [currentRecipe,setCurrentRecipe] = useState('')
+    const [url,setUrl] = useState(`https://api.spoonacular.com/recipes/complexSearch?query=${urlEndpoints.searchTerm}&number=${urlEndpoints.noOfResults}&offset=${urlEndpoints.offset}&cuisine=${urlEndpoints.cuisine}&diet=${urlEndpoints.diet}&intolorances=${urlEndpoints.intolorances}&type=${urlEndpoints.recipeType}&apiKey=033797df84694890b040b816a119b147`)
 
-    const handleGetUserData = ()=>{
-            getUserData(currentUser)
-    }
+    // HANDLE SET URL CALLBACK
+    const handleSetUrl = useCallback(()=>{
+        setUrl(`https://api.spoonacular.com/recipes/complexSearch?query=${urlEndpoints.searchTerm}&number=${urlEndpoints.noOfResults}&offset=${urlEndpoints.offset}&cuisine=${urlEndpoints.cuisine}&diet=${urlEndpoints.diet}&intolorances=${urlEndpoints.intolorances}&type=${urlEndpoints.recipeType}&apiKey=033797df84694890b040b816a119b147`)
+    },[urlEndpoints])
 
-    // getuserData on currentuser change
-    useEffect(()=>{
-        //could use void() here
-        signedIn && currentUser && handleGetUserData()
-    },[currentUser])
+    // GET RECIPES
```

Comment:
when using `useEffect()` with a dependency array that relies on state changing.. since you have to include the function in the dependency array if you include one in this case...
 is enclosing the function in `useCallback()` acceptable to prevent the `useEffect` from running the function on re-render since the function would be re-created for every render and cause an infinite loop? 

also on line `97` should the dependency array of that `useEffect()` include `url` since `getRecipes` uses it as a dependency or is that redundant.